finetuner/tuner/onnx.py

```diff
@@ -0,0 +1,206 @@
+
+    fm = get_framework(embed_model)
+
+    # Get framework-specific func to register model in ONNX
+    _parse_and_apply_to_onnx_func(fm)
+
+    _check_onnx_model(path)
+
+
+def _check_onnx_model(path: str) -> None:
+    """Check an ONNX model
+    :param path: Path to ONNX model
+    """
+    import onnx
+
+    model = onnx.load(path)
+    onnx.checker.check_model(model)
+
+
+def _to_onnx_torch(
+    embed_model: AnyDNN,
+    path: str,
+    input_shape: Tuple[int, ...],
+    opset_version: int = 11,
+    batch_size: int = 16,
+) -> None:
+    """Convert a PyTorch embedding model to the ONNX format
+    :param embed_model: Embedding model to register in ONNX
+    :param path: Patch where to register ONNX model to
+    :param input_shape: Embedding model input shape
+    :param batch_size: The batch size during export
+    :param opset_version: ONNX opset version in which to register
+    """
+
+    import torch
+
+    embed_model.eval()
+
+    x = torch.randn([batch_size] + input_shape, requires_grad=True, dtype=torch.float32)
+
+    torch.onnx.export(
+        embed_model,
+        x,
+        path,
+        export_params=True,
+        do_constant_folding=True,
+        opset_version=opset_version,
+        input_names=['input'],
+        output_names=['output'],
+        dynamic_axes={'input': {0: 'batch_size'}, 'output': {0: 'batch_size'}},
+    )
+
+
+def _to_onnx_keras(
+    embed_model: AnyDNN,
+    path: str,
+    input_shape: Tuple[int, ...],
+    opset_version: int = 11,
+) -> None:
+    """Convert a Keras embedding model to the ONNX format
+    :param embed_model: Embedding model to register in ONNX
+    :param path: Patch where to register ONNX model to
+    :param input_shape: Embedding model input shape
+    :param opset_version: ONNX opset version in which to register
+    """
+
+    try:
+        import tf2onnx
+    except (ImportError, ModuleNotFoundError):
+        raise ModuleNotFoundError('Module tf2onnx not found, try "pip install tf2onnx"')
+
+    import tensorflow as tf
+
+    shape = [
+        None,
+    ] + input_shape
+
+    _ = tf2onnx.convert.from_keras(
+        embed_model,
+        input_signature=[tf.TensorSpec(shape)],
+        opset=opset_version,
+        output_path=path,
+    )
+
+
+def _to_onnx_paddle(
+    embed_model: AnyDNN,
+    path: str,
+    input_shape: List[int],
+    opset_version: int = 11,
+) -> None:
+    """Convert a paddle embedding model to the ONNX format
+    :param embed_model: Embedding model to register in ONNX
+    :param path: Patch where to register ONNX model to
+    :param input_shape: Embedding model input shape
+    :param opset_version: ONNX opset version in which to register
+    """
+
+    # Removing onnx extension as paddle adds it automatically
+    if path.endswith(".onnx"):
+        path = path[:-5]
+
+    import paddle
+    from paddle.static import InputSpec
+
+    shape = [None] + list(input_shape)
+    x_spec = InputSpec(shape, 'float32', 'input')
```

Comment:
same, the `input_dtype` should be configurable

Reply:
Changed it here.